### PR TITLE
boards/slstk3400a: add support

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This PR adds support for the Silicon Labs [SLSTK3400A](https://www.silabs.com/development-tools/mcu/32-bit/efm32hg-starter-kit) Starter Kit.

### Testing procedure
The board is quite similar to the STK3200 and is based on the already supported EFM32 CPU.

I don't own this board, but @akshaim asked me to add this board. He is able to test this board.

### Issues/PRs references
Depends on #15299.

### CODE REVIEW

**Comment by benpicco** [comment]
You probably want to run `dist/tools/insufficient_memory/add_insufficient_memory_board.sh`.

**Comment by fjmolinas** [comment]
@akshaim can you provide some tests results?

File: cpu/efm32/families/efm32hg/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Find the header file that should exist if the CPU is supported. Only headers
+# for supported boards are included, but to support another CPU, it should be
+# as easy as adding the header file only.
+EFM32_HEADER = $(wildcard $(RIOTCPU)/efm32/families/efm32hg/include/vendor/$(CPU_MODEL).h)
+
+# include vendor device headers
+INCLUDES += -I$(RIOTCPU)/efm32/families/efm32hg/include/vendor
```
Comment (leandrolanzieri) [inline cpu/efm32/families/efm32hg/Makefile.include:7]: Missing newline

**Comment by leandrolanzieri** [comment]
I have the hardware, I'll run the script

File: cpu/efm32/families/efm32hg/Makefile.include (hunk 1 above)
Comment (basilfx) [reply cpu/efm32/families/efm32hg/Makefile.include]: Fixed and directly squashed.

File: boards/slstk3400a/Makefile.features [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CPU = efm32
```
[the remark this answers is not in the document]
Comment (basilfx) [reply boards/slstk3400a/Makefile.features:1]: I agree. I'll put it on my list :)

**Comment by basilfx** [comment]
Blacklisted a few more tests, and added 656dc3c to fix the MPU tests (it doesn't have a MPU).

Two tests will fail for the same reason as #15299 failed. It therefore depends on that PR to be merged first.

**Comment by leandrolanzieri** [comment]
Here I attached the results of running the tests: [results.zip](https://github.com/RIOT-OS/RIOT/files/5494668/results.zip)

The summary:
```
Failures during compilation:
- [examples/suit_update](examples/suit_update/compilation.failed)
- [tests/mpu_noexec_ram](tests/mpu_noexec_ram/compilation.failed)
- [tests/mpu_stack_guard](tests/mpu_stack_guard/compilation.failed)
- [tests/pkg_emlearn](tests/pkg_emlearn/compilation.failed)
- [tests/suit_manifest](tests/suit_manifest/compilation.failed)

Failures during test:
- [tests/cpp11_condition_variable](tests/cpp11_condition_variable/test.failed)
- [tests/cpp11_mutex](tests/cpp11_mutex/test.failed)
- [tests/cpp11_thread](tests/cpp11_thread/test.failed)
- [tests/driver_apds99xx](tests/driver_apds99xx/test.failed)
- [tests/driver_apds99xx_full](tests/driver_apds99xx_full/test.failed)
- [tests/driver_at86rf2xx_aes](tests/driver_at86rf2xx_aes/test.failed)
- [tests/driver_bme680](tests/driver_bme680/test.failed)
- [tests/driver_hd44780](tests/driver_hd44780/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/pkg_flatbuffers](tests/pkg_flatbuffers/test.failed)
- [tests/pkg_littlefs](tests/pkg_littlefs/test.failed)
- [tests/pkg_semtech-loramac](tests/pkg_semtech-loramac/test.failed)
- [tests/pthread_barrier](tests/pthread_barrier/test.failed)
- [tests/pthread_cooperation](tests/pthread_cooperation/test.failed)
- [tests/riotboot](tests/riotboot/test.failed)
```

The emlearn is not building because I'm lacking some python packages.

**Comment by basilfx** [comment]
Thanks @leandrolanzieri. I will take a look at the results tonight.

**Comment by akshaim** [comment]
Thanks, @basilfx  @leandrolanzieri  for the help :) Was away for a while. Regarding the tests, I had tested the board's onboard GPIO, it worked fine though.

File: boards/slstk3400a/Makefile.features (hunk 2 above)
[the remark this answers is not in the document]
Comment (basilfx) [reply boards/slstk3400a/Makefile.features:1]: Should be fixed with #15425.

**Comment by aabadie** [comment]
> Here I attached the results of running the tests

If you build in docker, you will have all missing dependencies.

File: boards/slstk3400a/Makefile.include [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
```
Comment (aabadie) [inline boards/slstk3400a/Makefile.include:6]: All this can now be removed.

Comment (basilfx) [reply boards/slstk3400a/Makefile.include]: Done.

**Comment by basilfx** [comment]
Rebased and updated the PR.

**Comment by benpicco** [comment]
CI found some issues

**Comment by basilfx** [comment]
Fixed the issues (more CI ignores) directly. Also rebased it.

**Review by benpicco — APPROVED** [review]
Looks good to me, another step at getting all those auto-generated EFM32 MCUs mainlined.

The failed tests hint at either missing memory for dynamic allocation (board only has 8k MCU) or missing hardware.

There should be no surprises since this only uses common EFM32 code.

**Pull request merged**